ui/src/routes/_layout/organizations/$orgId/products/$productId/repositories/$repoId/users/index.tsx

```diff
@@ -0,0 +1,253 @@
+ * License-Filename: LICENSE
+ */
+
+import { zodResolver } from '@hookform/resolvers/zod';
+import { createFileRoute } from '@tanstack/react-router';
+import { Loader2 } from 'lucide-react';
+import { useForm } from 'react-hook-form';
+import { z } from 'zod';
+
+import {
+  useGroupsServiceDeleteUserFromGroupRepository,
+  useGroupsServicePutUserToGroupRepository,
+} from '@/api/queries';
+import { prefetchUseRepositoriesServiceGetRepositoryById } from '@/api/queries/prefetch';
+import { useRepositoriesServiceGetRepositoryByIdSuspense } from '@/api/queries/suspense';
+import { ApiError } from '@/api/requests';
+import { ToastError } from '@/components/toast-error';
+import { Button } from '@/components/ui/button';
+import {
+  Card,
+  CardContent,
+  CardDescription,
+  CardFooter,
+  CardHeader,
+  CardTitle,
+} from '@/components/ui/card';
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from '@/components/ui/form';
+import { Input } from '@/components/ui/input';
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from '@/components/ui/select';
+import { toast } from '@/lib/toast';
+import { groupsSchema } from '@/schemas';
+
+const formSchema = z.object({
+  username: z.string().min(1),
+  groupId: groupsSchema,
+});
+
+const ManageUsers = () => {
+  const params = Route.useParams();
+
+  const { data: repository } = useRepositoriesServiceGetRepositoryByIdSuspense({
+    repositoryId: Number.parseInt(params.repoId),
+  });
+
+  const { mutateAsync: addUser, isPending: isAddUserPending } =
+    useGroupsServicePutUserToGroupRepository({
+      onSuccess() {
+        toast.info('Add User', {
+          description: `User "${form.getValues().username}" added successfully to group "${form.getValues().groupId.toUpperCase()}".`,
+        });
+      },
+      onError(error: ApiError) {
+        toast.error(error.message, {
+          description: <ToastError error={error} />,
+          duration: Infinity,
+          cancel: {
+            label: 'Dismiss',
+            onClick: () => {},
+          },
+        });
+      },
+    });
+
+  const { mutateAsync: removeUser, isPending: isRemoveUserPending } =
+    useGroupsServiceDeleteUserFromGroupRepository({
+      onSuccess() {
+        toast.info('Remove User', {
+          description: `User "${form.getValues().username}" removed successfully from group "${form.getValues().groupId.toUpperCase()}".`,
+        });
+      },
+      onError(error: ApiError) {
+        toast.error(error.message, {
+          description: <ToastError error={error} />,
+          duration: Infinity,
+          cancel: {
+            label: 'Dismiss',
+            onClick: () => {},
+          },
+        });
+      },
+    });
+
+  const form = useForm<z.infer<typeof formSchema>>({
+    resolver: zodResolver(formSchema),
+    defaultValues: {
+      username: '',
+      groupId: 'readers',
+    },
+  });
```

Comment:
This would be a little easier to follow if the form was initialized above the mutation hooks, as form is used in their toasts.